In [144]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sna



##### Uploading Data into df Variable

In [145]:
data = pd.read_csv('C:/Users/sitan/OneDrive/Desktop/Data Science/Python-14-Feb/DA_Python/movies.csv')

##### Showing Data

In [146]:
Movie_Data = pd.DataFrame(data)
Movie_Data

MOVIES         YEAR  \
0                           Blood Red Sky       (2021)   
1     Masters of the Universe: Revelation     (2021– )   
2                        The Walking Dead  (2010–2022)   
3                          Rick and Morty     (2013– )   
4                         Army of Thieves       (2021)   
...                                   ...          ...   
9994                       The Imperfects     (2021– )   
9995                               Arcane     (2021– )   
9996                    Heart of Invictus     (2022– )   
9997                       The Imperfects     (2021– )   
9998                       The Imperfects     (2021– )   

                                           GENRE  RATING  \
0         \nAction, Horror, Thriller                 6.1   
1     \nAnimation, Action, Adventure                 5.0   
2          \nDrama, Horror, Thriller                 8.2   
3     \nAnimation, Adventure, Comedy                 9.2   
4            \nAction, Crime, Horror                 NaN   
...                                          ...     ...   
9994     \nAdventure, Drama, Fantasy                 NaN   
9995  \nAnimation, Action, Adventure                 NaN   
9996            \nDocumentary, Sport                 NaN   
9997     \nAdventure, Drama, Fantasy                 NaN   
9998     \nAdventure, Drama, Fantasy                 NaN   

                                               ONE-LINE  \
0     \nA woman with a mysterious illness is forced ...   
1     \nThe war for Eternia begins again in what may...   
2     \nSheriff Deputy Rick Grimes wakes up from a c...   
3     \nAn animated series that follows the exploits...   
4     \nA prequel, set before the events of Army of ...   
...                                                 ...   
9994                                     \nAdd a Plot\n   
9995                                     \nAdd a Plot\n   
9996                                     \nAdd a Plot\n   
9997                                     \nAdd a Plot\n   
9998                                     \nAdd a Plot\n   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Gross  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN  
...    ...  
9994   NaN  
9995   NaN  
9996   NaN  
9997   NaN  
9998   NaN  

[9999 rows x 9 columns]

In [147]:
Movie_Data.sample(10)

MOVIES         YEAR  \
6485             Avatar: The Last Airbender  (2005–2008)   
6853                        The Pentaverate         (II)   
7005   La Reina de Indias y el Conquistador     (2020– )   
5363            Hong Kong West Side Stories   (I) (2019)   
8212                        Black Lightning  (2017–2021)   
1754                                  Masum       (2017)   
1476                           Premium Rush       (2012)   
8411                  El desorden que dejas       (2020)   
4334                      #TemanTapiMenikah       (2018)   
946                               Greenleaf  (2016–2020)   

                                           GENRE  RATING  \
6485  \nAnimation, Action, Adventure                 9.6   
6853                        \nComedy                 NaN   
7005                \nDrama, History                 NaN   
5363                 \nComedy, Drama                 7.3   
8212         \nAction, Drama, Sci-Fi                 7.3   
1754         \nCrime, Drama, Mystery                 8.4   
1476       \nAction, Crime, Thriller                 6.5   
8411         \nCrime, Drama, Mystery                 7.0   
4334      \nBiography, Comedy, Drama                 6.9   
946                          \nDrama                 7.4   

                                               ONE-LINE  \
6485  \nAs Zuko tries to make it by in exile without...   
6853                                     \nAdd a Plot\n   
7005  \nDuring the council's vote on whether to oust...   
5363  \nA dark comedy reflecting truth of happening ...   
8212  \nAfter learning that Gravedigger and the Mark...   
1754  \nCevdet (Haluk Bilginer), a retired commissio...   
1476  \nIn Manhattan, a bike messenger picks up an e...   
8411  \nRoi decides to do some investigating of his ...   
4334  \nAyudia (Vanesha Prescilla) and Ditto (Adipat...   
946   \nThe unscrupulous world of the Greenleaf fami...   

                                                  STARS    VOTES  RunTime  \
6485  \n    Director:\nLauren MacMullan\n| \n    Sta...    4,263     25.0   
6853  \n    Director:\nTim Kirkby\n| \n    Stars:\nD...      NaN      NaN   
7005  \n            \n    Stars:\nEmmanuel Esparza, ...      NaN      NaN   
5363  \n            \n    Stars:\nLouis Cheung, \nSi...       82      NaN   
8212  \n    Director:\nOz Scott\n| \n    Stars:\nCre...      300     41.0   
1754  \n            \n    Stars:\nHaluk Bilginer, \n...   16,763     60.0   
1476  \n    Director:\nDavid Koepp\n| \n    Stars:\n...  111,533     91.0   
8411  \n    Director:\nRoger Gual\n| \n    Stars:\nI...      153     42.0   
4334  \n    Director:\nRako Prijanto\n| \n    Stars:...      602    102.0   
946   \n            \n    Stars:\nMerle Dandridge, \...    4,624     60.0   

        Gross  
6485      NaN  
6853      NaN  
7005      NaN  
5363      NaN  
8212      NaN  
1754      NaN  
1476  $20.28M  
8411      NaN  
4334      NaN  
946       NaN

##### Finding Duplicate Row

In [148]:
duplicates = Movie_Data[Movie_Data.duplicated()]
duplicates


MOVIES      YEAR  \
6833   Mighty Little Bheem  (2019– )   
6835   Mighty Little Bheem  (2019– )   
6836   Mighty Little Bheem  (2019– )   
6837   Mighty Little Bheem  (2019– )   
6838   Mighty Little Bheem  (2019– )   
...                    ...       ...   
9989                  1899  (2022– )   
9990                  1899  (2022– )   
9991                  1899  (2022– )   
9992                  1899  (2022– )   
9994        The Imperfects  (2021– )   

                                          GENRE  RATING        ONE-LINE  \
6833  \nAnimation, Short, Adventure                 NaN  \nAdd a Plot\n   
6835  \nAnimation, Short, Adventure                 9.0  \nAdd a Plot\n   
6836  \nAnimation, Short, Adventure                 9.0  \nAdd a Plot\n   
6837  \nAnimation, Short, Adventure                 NaN  \nAdd a Plot\n   
6838  \nAnimation, Short, Adventure                 NaN  \nAdd a Plot\n   
...                                         ...     ...             ...   
9989       \nDrama, History, Horror                 NaN  \nAdd a Plot\n   
9990       \nDrama, History, Horror                 NaN  \nAdd a Plot\n   
9991       \nDrama, History, Horror                 NaN  \nAdd a Plot\n   
9992       \nDrama, History, Horror                 NaN  \nAdd a Plot\n   
9994    \nAdventure, Drama, Fantasy                 NaN  \nAdd a Plot\n   

                                                  STARS VOTES  RunTime Gross  
6833  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...   NaN      NaN   NaN  
6835  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...     6      NaN   NaN  
6836  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...     6      NaN   NaN  
6837  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...   NaN      NaN   NaN  
6838  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...   NaN      NaN   NaN  
...                                                 ...   ...      ...   ...  
9989                   \n    Director:\nBaran bo Odar\n   NaN      NaN   NaN  
9990                   \n    Director:\nBaran bo Odar\n   NaN      NaN   NaN  
9991                   \n    Director:\nBaran bo Odar\n   NaN      NaN   NaN  
9992                   \n    Director:\nBaran bo Odar\n   NaN      NaN   NaN  
9994  \n            \n    Stars:\nMorgan Taylor Camp...   NaN      NaN   NaN  

[431 rows x 9 columns]

##### Before Removing Duplicate total Number of Rows is 9999. After Removing Duplicate Rows 9568. Total Number of Duplicate Row which is removed 431

In [149]:
Movie_Data.drop_duplicates(inplace=True,ignore_index=True)

##### Cleaning Year Column

In [150]:
Movie_Data['YEAR'] = Movie_Data['YEAR'].str.extract(r'(\d{4})')
Movie_Data['YEAR']

0       2021
1       2021
2       2010
3       2013
4       2021
        ... 
9563    2022
9564    2021
9565    2022
9566    2021
9567    2021
Name: YEAR, Length: 9568, dtype: object

In [151]:
Movie_Data[(Movie_Data['YEAR'].isna()) & (Movie_Data['YEAR']!=None)][['MOVIES','YEAR']]

MOVIES YEAR
878   Vikings: Valhalla  NaN
1155         The Killer  NaN
1165             Hustle  NaN
1375      Resident Evil  NaN
1414           Bright 2  NaN
...                 ...  ...
9515       Pretty Smart  NaN
9516       Pretty Smart  NaN
9517       Pretty Smart  NaN
9518       Pretty Smart  NaN
9519       Pretty Smart  NaN

[635 rows x 2 columns]

In [152]:
Movie_Data['STARS']=Movie_Data['STARS'].str.replace("\n","")

In [153]:
Movie_Data['ONE-LINE']=Movie_Data['ONE-LINE'].str.replace("\n"," ")

In [ ]:
# Dictionary of movie index and release year
movie_years = {
    6243: 2025,   # Untitled Sean O'Keefe Project
    6244: 2025,   # Perfectly Legal
    6246: 2025,   # Change Agent
    6249: 2025,   # Sharing the Wisdom of Time
    6250: 2025,   # Nueva serie de Daniel Sánchez Arévalo
    6252: 2025,   # Yuba
    6253: 2025,   # Quicksand
    6256: 2025,   # Signs
    6260: 2025,   # Untitled U.S. Women's Soccer Team Project
    6264: 2025,   # The Hall: Honoring the Greats of Stand Up
    6269: 2025,   # Open 24 7
    6277: 2025,   # The Last
    6280: 2025,   # Kapil Sharma Special
    6283: 2025,   # Untitled Tom Segura Netflix Special
    6285: 2025,   # Untitled Elodie Polo Ackerman Netflix Docuseries
    6290: 2025,   # IRL: In Real Love
    6295: 2025,   # House Arrest
    6297: 2025,   # Untitled Netflix Billy Eichner Stand up Special
    6302: 2025,   # Juanpis
    6303: 2025,   # Untitled Patrick Osborne Fantasy Project
    6305: 2025,   # Reset: My fight for Inclusion and Lasting Change
    6310: 2025,   # Big Vape: The Incendiary Rise of Juul
    6313: 2025,   # Amazing On The Inside
    6318: 2025,   # Untitled Netflix BBC Science Documentary Project
    6319: 2025,   # Hot Skull
    6321: 2025,   # Alejandra Azcárate
    6323: 2025,   # Antiracist Baby
    6326: 2025,   # The Ring of Fire
    6328: 2025,   # Johnny by Johnny
    6329: 2025,   # Untitled Akin Omotoso Netflix Project
    6332: 2025,   # Canciones y Recuerdos
    6335: 2025,   # Freaks
    6336: 2025,   # Sumukhi Suresh Special
    6337: 2025,   # Isabella's Witch Chronicles
    6345: 2025,   # Wraith
    6347: 2025,   # Temporada de Verão
    6349: 2025,   # Life Sentence
    6351: 2025,   # Grande Terim
    6352: 2025,   # Far Cry
    6353: 2025,   # The 7 lives of Lea
    6355: 2025,   # Stamped: Racism, Antiracism and You
    6360: 2025,   # Oceans
    6363: 2025,   # Queen of the Scene
    6366: 2025,   # Wild Guys
    6368: 2025,   # Untitled Sefi Atta Project
    6371: 2025,   # Summer in Sorrento
    6378: 2025,   # Little Baby Bum: Music Time
    6383: 2025,   # Death Smiles
    6384: 2025,   # Life in Threes
    6388: 2025,   # Glória
    6390: 2025,   # Comfort Me with Apples
    6391: 2025,   # Exatlon Challenge
    6393: 2025,   # Connected
    6411: 2025,   # Extracurricular
    6413: 2025,   # Janguea y Rapea
    6414: 2025,   # What To Doodles
    6418: 2025,   # Untitled Woodstock Netflix Project
    6433: 2025,   # The Rat
    6437: 2025,   # Curtain
    6439: 2025,   # Untitled Jon Berg Netflix Project
    6449: 2025,   # Untitled Michael Diliberti Netflix Project
    6453: 2025,   # Untitled Tituss Burgess Netflix Project
    6455: 2025,   # El Negocio, Pt. II
    6457: 2025,   # Untitled Hend Sabry Project
    6470: 2025,   # The Kane Chronicles: The Red Pyramid
    6471: 2025,   # The Kane Chronicles: The Throne of Fire
    6472: 2025,   # The Kane Chronicles: The Serpent's Shadow
    6836: 2025,   # The Pentaverate
    6837: 2025,   # The Pentaverate
    6838: 2025,   # The Pentaverate
    6839: 2025,   # The Pentaverate
    6840: 2025,   # The Pentaverate
    8144: 2025,   # One Piece
    8149: 2025,   # One Piece
    8249: 2025,   # Painkiller
    8250: 2025,   # Painkiller
    8312: 2025,   # In from the Cold
    8313: 2025,   # In from the Cold
    8314: 2025,   # In from the Cold
    8315: 2025,   # In from the Cold
    8316: 2025,   # In from the Cold
    8317: 2025,   # In from the Cold
    8318: 2025,   # In from the Cold
    8475: 2025,   # Baahubali: Before the Beginning
    8598: 2025,   # Jaguar
    8746: 2025,   # The Witcher: Blood Origin
    8747: 2025,   # The Witcher: Blood Origin
    8780: 2025,   # Jaguar
    8784: 2025,   # Anatomy of a Scandal
    8800: 2025,   # You Don't Know Me
    8801: 2025,   # You Don't Know Me
    8802: 2025,   # You Don't Know Me
    8803: 2025,   # You Don't Know Me
    8825: 2025,   # Vikings: Valhalla
    8826: 2025,   # Vikings: Valhalla
    8837: 2025,   # Resident Evil
    8880: 2025,   # Vikings: Valhalla
    8913: 2025,   # Clark
    8921: 2025,   # Vikings: Valhalla
    8922: 2025,   # Vikings: Valhalla
    8931: 2025,   # Vikings: Valhalla
    8932: 2025,   # Vikings: Valhalla
    8933: 2025,   # Vikings: Valhalla
    8934: 2025,   # Vikings: Valhalla
    8935: 2025,   # Vikings: Valhalla
    8936: 2025,   # Vikings: Valhalla
    8937: 2025,   # Vikings: Valhalla
    8938: 2025,   # Vikings: Valhalla
    8959: 2025,   # Lost Ollie
    8965: 2025,   # Clark
    8966: 2025,   # Clark
    8967: 2025,   # Clark
    8970: 2025,   # Feria: The Darkest Light
    9043: 2025,   # Stay Close
    9044: 2025,   # Stay Close
    9045: 2025,   # Stay Close
    9046: 2025,   # Stay Close
    9054: 2025,    # Archive 81
    6078: 2025,   # Sweetener
    6080: 2025,   # A Note of Explanation
    6083: 2025,   # Invisible City
    6084: 2025,   # Untitled Eddie Murphy Project
    6097: 2025,   # Grimms' Fairy Tales
    6100: 2025,   # One Piece: Entering into the Grand Line
    6101: 2025,   # Holdout
    6102: 2025,   # The Murders of Molly Southbourne
    6109: 2025,   # The Stronghold
    6111: 2025,   # A god in ruins
    6113: 2025,   # Lost Dog
    6123: 2025,   # The F U
    6125: 2025,   # Untitled Haiti Musical
    6126: 2025,   # Maska
    6130: 2025,   # The Ice Bucket Challenge
    6136: 2025,   # Hunch
    6137: 2025,   # The Boys' Club
    6138: 2025,   # Matilda
    6139: 2025,   # Running with Sherman
    6140: 2025,   # Three Sixty (360)
    6143: 2025,   # My Dearest, Like a Cherry Blossom
    6144: 2025,   # Shadecraft
    6154: 2025,   # Prodigy
    6159: 2025,   # Arranged Marriage
    6162: 2025,   # Green Cross
    6175: 2025,   # Eres Tú
    6176: 2025,   # It's Wednesday Night
    6177: 2025,   # To Hell with the Devil
    6181: 2025,   # Again
    6185: 2025,   # Legends of Strength
    6187: 2025,   # Untitled Prentice Penny Holiday Project
    6190: 2025,   # Most Likely To
    6195: 2025,   # Bardo
    6202: 2025,   # Happiness Ever After
    6206: 2025,   # Dark Days at the Magna Carta
    6208: 2025,   # Wild Rythm
    6211: 2025,   # Untitled Bubba Wallace Netflix Project
    6215: 2025,   # Lost Bullet Sequel
    6216: 2025,   # All the Right Moves
    6218: 2025,   # Untitled Shady El Hamus Hip Hop Project
    6219: 2025,   # The Poet
    6233: 2025,   # The Get Away
    6239: 2025,    # Byron Baes
    5390: 2022,   # Untitled Andy Warhol Netflix Documentary Project
    5402: 2025,   # Student Driver
    5412: 2025,   # Last Sons of America
    5420: 2025,   # Seoul Daejakjeon
    5429: 2025,   # Exception
    5447: 2025,   # The Witch Boy
    5449: 2022,   # Wednesday
    5456: 2025,   # Suriname
    5472: 2025,   # Untitled Bird Box Spanish Spinoff
    5485: 2022,   # Kleo
    5487: 2025,   # The Unsound
    5499: 2025,   # Shout It Out Loud
    5528: 2025,   # Redd Zone
    5534: 2020,   # Paranormal
    5543: 2025,   # Auntie Claus
    5546: 2025,   # Conquest
    5558: 2025,   # Blackout
    5564: 2025,   # Baby Reindeer
    5584: 2025,   # This Perfect Day
    5590: 2025,   # Untitled Solvan Naim Project
    5603: 2025,   # The Standoff
    5612: 2025,   # Fish in a Tree
    5613: 2025,   # American Jesus
    5614: 2025,   # The Fifth Risk
    5615: 2025,   # Caste
    5617: 2025,   # Harvey
    5620: 2022,   # Le Monde de demain
    5630: 2025,   # Ghost in the Machine
    5644: 2025,   # Ripple
    5649: 2025,   # I Am Not Your Perfect Mexican Daughter
    5650: 2025,   # Meet the Voxels
    5665: 2025,   # Kingdom Come
    5667: 2025,   # Bloom
    5674: 2025,   # Untitled We are the Nobles Reboot
    5676: 2025,   # The Warmth of Other Suns
    5681: 2025,   # Dennis Nilsen's the History of A Drowning Boy
    5683: 2025,   # Tú no eres especial
    5685: 2025,   # Untitled Castlevania Project
    5686: 2025,   # The Hills Have Eyes for You
    5700: 2025,   # Supercrooks
    5705: 2025,   # Untitled Spenser Confidential Sequel
    5736: 2025,   # Free Food for Millionaires
    5751: 2025,   # Untitled Charlize Theron Netflix Project
    5758: 2025,   # Transformers: BotBots
    5762: 2025,   # Heaven's Forest
    5764: 2025,   # Dead of Winter
    5766: 2025,   # Untitled Nipsey Hussle Documentary
    5775: 2025,   # The Residence
    5778: 2022,   # Uncoupled
    5780: 2025,   # Pico & Sepulveda
    5783: 2025,   # The Chameleon
    5784: 2025,   # Tunga
    5787: 2019,   # Home for Christmas
    5789: 2025,   # Idol
    5792: 2025,   # The City of Brass
    5796: 2025,   # The Cactus
    5797: 2025,   # Ídolo
    5802: 2025,   # Sharkey the Bounty Hunter
    5804: 2025,   # Untitled Netflix My Little Pony Series
    5805: 2025,   # Untitled Queen Charlotte Project
    6002: 2022,   # If Only
    6003: 2015,   # Bhaage Re Mann
    6005: 2023,   # Big Vape
    6014: 2025,   # Untitled Will Smith Comedy Variety Special
    6016: 2025,   # Mech Cadets
    6021: 2025,   # Selfie
    6023: 2025,   # Untitled Conan Netflix Project
    6026: 2025,   # Shimmers
    6028: 2025,   # Stamped from the Beginning
    6034: 2025,   # Untitled Kenya Barris Netflix Project
    6047: 2025,   # Goddess Girls
    6048: 2025,   # Forty Acres
    6050: 2025,   # The Tradition
    6053: 2019,   # Daybreak
    6058: 2025,   # Untitled Vikram Motwane Project
    6062: 2025,   # Call Your Daughter Home
    6066: 2016,    # The BFG
    5808: 2022,   # GameStop
    5809: 2022,   # Thermae Romae Novae
    5813: 2025,   # Far Cry
    5816: 2025,   # Ghee Happy
    5818: 2025,   # Steps
    5820: 2025,   # Amnesty
    5825: 2025,   # The Outlaw Ocean
    5828: 2025,   # Pokémon
    5834: 2025,   # The Red Zone
    5839: 2022,   # Indian Predator
    5842: 2025,   # Juneteenth
    5843: 2025,   # Dino Daycare
    5844: 2025,   # First Ascent
    5849: 2025,   # Game Changer
    5850: 2022,   # Matilda
    5856: 2025,   # Untitled Fernando Meirelles Netflix Project
    5859: 2025,   # Hummingbird Salamander
    5860: 2025,   # Untitled Virunga Project
    5868: 2025,   # Parallel
    5870: 2014,   # Tusk
    5874: 2025,   # Notes on Love
    5879: 2025,   # Sunshine Scouts
    5899: 2022,   # Scarlet
    5901: 2025,   # Hello, Universe
    5902: 2022,   # Maldivas
    5903: 2025,   # Untitled Colton Underwood Netflix Project
    5905: 2025,   # The Hunt for Atlantis
    5906: 2022,   # Das Privileg
    5912: 2025,   # Line Friends
    5913: 2025,   # Free To Be You And Me
    5919: 2021,   # Crime Stories: India Detectives
    5921: 2025,   # Untitled Madhuri Dixit Netflix Project
    5923: 2022,   # Untitled Kanye West Documentary
    5926: 2025,   # Three Musketeers
    5931: 2025,   # Oni
    5936: 2022,   # Intimacy
    5941: 2025,   # Make My Day
    5944: 2022,   # Untitled LA Lakers Netflix Project
    5946: 2025,   # Stamped from the Beginning
    5947: 2022,   # Untitled Thai Cave Rescue Series
    5958: 2025,   # Sharp
    5961: 2025,   # I Take It Black
    5963: 2022,   # Erynie
    5964: 2025,   # The 50th Law
    5966: 2025,   # Janet
    5970: 2025,   # Untitled Jane Lynch Cyndi Lauper Netflix Project
    5971: 2025,   # Sulwe
    5980: 2025,   # A Life in Men
    5994: 2021,   # How to Be A Cowboy
    5995: 2025,   # Cocaine Hippos
    5996: 2025,   # Midnight's Children
    5997: 2025,   # Untitled Asterix Project
    5999: 2025,    # Hello America
    5270: 2022,   # Andropoz
    5271: 2025,   # Untitled We Can Be Heroes Sequel
    5276: 2025,   # Freedom
    5279: 2021,   # Mio fratello mia sorella
    5287: 2021,   # 13 Minutes
    5288: 2022,   # 42 días en la oscuridad
    5305: 2021,   # The Claus Family 2
    5319: 2025,   # On Her Own Ground
    5327: 2025,   # Bubbles
    5336: 2025,   # A Time Lost
    5341: 2025,   # Boons and Curses
    5357: 2022,   # Hometown
    5362: 2025,   # The Twits
    5373: 2022,   # Mochileros
    5375: 2025,   # Untitled Secret Service
    5384: 2025,    # Bone
    5180: 2021,   # Kitz
    5185: 2022,   # Tinder Swindler
    5199: 2025,   # Empress
    5202: 2025,   # Way Station
    5205: 2025,   # Watergate
    5219: 2022,   # Sen Yasamaya Bak
    5225: 2022,   # Spirit Rangers
    5232: 2025,   # Mystery Girl
    5244: 2025,   # Cuckoo Song
    5247: 2025,   # Pashmina
    5250: 2025,   # Untitled To All the Boys Spinoff
    5253: 2025,   # Mulligan
    5266: 2025,    # Huck
    5047: 2023,   # Agent King
    5048: 2025,   # Reborn
    5061: 2025,   # The Girls on the Bus
    5068: 2025,   # The Howling
    5072: 2023,   # Captain Fall
    5078: 2025,   # Astérix
    5085: 2025,   # À tombeau ouvert
    5086: 2022,   # Untitled Joe Exotic Project
    5089: 2025,   # Dorothy and Alice
    5091: 2025,   # Shoe Dog
    5103: 2025,   # Uprising
    5119: 2025,   # Ultraman
    5133: 2025,   # Tell No One
    5153: 2025,   # The Overstory
    5169: 2025,    # Sword Art Online
    9079: 2025,   # Unrecorded Night
    9141: 2022,   # First Kill
    9142: 2022,   # First Kill
    9182: 2022,   # Anatomy of a Scandal
    9183: 2022,   # Anatomy of a Scandal
    9184: 2022,   # Anatomy of a Scandal
    9185: 2022,   # Anatomy of a Scandal
    9194: 2021,   # True Story
    9195: 2022,   # Man vs. Bee
    9200: 2022,   # The Lincoln Lawyer
    9201: 2022,   # The Lincoln Lawyer
    9202: 2022,   # The Lincoln Lawyer
    9203: 2022,   # The Lincoln Lawyer
    9204: 2022,   # The Lincoln Lawyer
    9205: 2022,    # The Lincoln Lawyer
    9206: 2022,   # The Lincoln Lawyer
    9207: 2022,   # The Lincoln Lawyer
    9208: 2022,   # The Lincoln Lawyer
    9209: 2022,   # The Lincoln Lawyer
    9210: 2022,   # Bir Denizalti Hikayesi
    9218: 2022,   # Heartstopper
    9219: 2022,   # Heartstopper
    9238: 2022,   # The Last Bus
    9244: 2022,   # First Kill
    9245: 2022,   # First Kill
    9268: 2022,   # The Last Bus
    9269: 2022,   # The Last Bus
    9270: 2022,   # The Last Bus
    9271: 2022,   # The Last Bus
    9287: 2022,   # Angry Birds: Summer Madness
    9294: 2022,   # God's Favorite Idiot
    9341: 2021,   # True Story
    9342: 2021,   # True Story
    9343: 2021,   # True Story
    9347: 2022,   # Yeh Kaali Kaali Ankhein
    9350: 2022,    # Yeh Kaali Kaali Ankhein
    9397: 2022,   # From Scratch
    9426: 2022,   # Inside Man
    9443: 2022,   # From Scratch
    9444: 2022,   # From Scratch
    9445: 2022,   # From Scratch
    9468: 2022,   # From Scratch
    9476: 2022,   # Kus uçusu
    9486: 2022,   # Inside Man
    9515: 2021,   # Pretty Smart
    9516: 2021,   # Pretty Smart
    9517: 2021,   # Pretty Smart
    9518: 2021,   # Pretty Smart
    9519: 2021,    # Pretty Smart
    4902: 2025,   # Sexy Central
    4905: 2025,   # Braqueurs
    4909: 2021,   # An Astrological Guide for Broken Hearts
    4911: 2021,   # Karma's World
    4917: 2022,   # Mr. Harrigan's Phone
    4919: 2022,   # Pearl
    4920: 2025,   # Skull Island
    4925: 2022,   # Notre Dame, la Part du feu
    4940: 2025,   # Talisman
    4945: 2025,   # Untitled Netflix Chronicles of Narnia Series
    4948: 2022,   # The G Word with Adam Conover
    4967: 2025,   # Untitled Netflix Kevin James Project
    4974: 2025,   # I Believe in A Thing Called Love
    4979: 2022,   # Il filo invisibile
    5023: 2025,   # My Wife and I Bought A Ranch...
    5038: 2025,   # The Ogun
    5040: 2025,    # Untitled Netflix Chronicles of Narnia TV Movie
    2978: 2021,   # Pretty Smart
    2980: 2023,   # Heart of Stone
    2994: 2022,   # Samurai Rabbit: The Usagi Chronicles
    3014: 2022,   # BigBug
    3017: 2022,   # Wedding Season
    3062: 2023,   # Chupa
    3165: 2022,    # Untitled Ersan Kuneri Project
    3197: 2023,   # Beef
    3210: 2022,   # Yeh Kaali Kaali Ankhein
    3216: 2020,   # Most Dangerous Game
    3217: 2025,   # Untitled Arnold Schwarzenegger Skydance Project
    3230: 2024,   # The Three Body Problem
    3248: 2023,   # My Dad the Bounty Hunter
    3255: 2022,   # Ivy & Bean
    3307: 2023,   # Upstate
    3314: 2024,   # Breathe
    3326: 2025,   # Untitled Justin Hartley Netflix Project
    3328: 2025,   # The Selection
    3334: 2025,   # First Ladies
    3336: 2025,    # Les Italiens
    3338: 2025,   # Magic: The Gathering
    3356: 2021,   # Mixtape
    3374: 2022,   # An Afternoon with SCTV
    3375: 2022,   # The Weekend Away
    3433: 2023,   # The Upper World
    3437: 2024,   # Gordon Hemingway & The Realm of Cthulhu
    3440: 2023,   # Blue Eye Samurai
    3462: 2025,   # Chinatown
    3471: 2025,   # Untitled Netflix Mark Boal Project
    3477: 2025,   # Splinter Cell
    3502: 2024,   # Damsel
    3520: 2025,    # You Don't Know Me
    3522: 2022,   # Kus uçusu
    3565: 2025,   # The Talisman
    3577: 2022,   # In from the Cold
    3596: 2022,   # A Través De Mi Ventana
    3610: 2022,   # Kolejne 365 dni
    3616: 2025,   # Wings of Fire
    3617: 2021,   # The Unlikely Murderer
    3621: 2022,   # The Last Bus
    3655: 2022,   # Money Heist Korea
    3686: 2023,   # Rustin
    3689: 2022,   # The Outlaws
    3693: 2025,   # Untitled Nora Roberts Project
    3707: 2025,   # White Stork
    3729: 2025,    # Lady Killer
    3739: 2025,   # Flash Boys
    3764: 2025,   # Untitled Jason Bateman John Cena Netflix Project
    3778: 2025,   # The Chronicles of Narnia: The Magician's Nephew
    3804: 2025,   # Carmen Sandiego
    3837: 2025,   # Family Leave
    3858: 2025,   # The Formula
    3872: 2025,   # Hamlet
    3880: 2025,   # Baahubali: Before the Beginning
    3892: 2022,   # Angry Birds: Summer Madness
    3905: 2023,   # The Night Agent
    3908: 2025,   # Animal Farm
    3912: 2025,   # The Helicopter Heist
    3916: 2025,   # This Beast
    3934: 2025,   # Untitled Vince Vaughn Netflix Project
    3935: 2025,    # Alice in Wonderland
    3945: 2021,   # Aranyak
    3951: 2025,   # Six Years
    3984: 2025,   # Super Normal
    3989: 2025,   # Jaguar
    3994: 2023,   # The Out-Law
    4019: 2021,   # Bright: Samurai Soul
    4021: 2025,   # The Magic Order
    4039: 2025,   # Oompa Loompas
    4083: 2025,   # The Last Will and Testament of Charles Abernathy
    4109: 2022,   # Érase una vez... pero ya no
    4152: 2025,   # The Warriors
    4168: 2025,   # Ask Taktikleri
    4170: 2025,   # A Chorus Line
    4172: 2025,   # We Used to Live Here
    4173: 2022,   # Si Lo Hubiera Sabido
    4190: 2021,   # Anxious People
    4202: 2025,    # Thelma the Unicorn
    4209: 2025,   # Lady Business
    4221: 2025,   # Something is Killing the Children
    4235: 2022,   # Annarasumanara
    4237: 2022,   # Rilakkuma's Theme Park Adventure
    4250: 2025,   # The Bellhop
    4279: 2025,   # Anna Karenina
    4281: 2022,   # Monica O My Darling
    4288: 2025,   # Untitled Skydance TV's hourlong series
    4292: 2025,   # The Cipher
    4294: 2021,   # El tiempo que te doy
    4298: 2023,   # Your Place or Mine
    4311: 2025,   # Mega Man
    4312: 2022,   # Pera Palas'ta Gece Yarisi
    4315: 2025,   # Black Stallions
    4335: 2025,    # Kaos
    4337: 2022,   # Untitled Thai Cave Rescue Project
    4347: 2025,   # The Legend of Cocaine Island
    4358: 2025,   # Untitled Gundam Project
    4360: 2025,   # Good Times
    4371: 2025,   # Krishna Balram
    4398: 2021,   # A Tale Dark and Grimm
    4419: 2021,   # Decoupled
    4436: 2022,   # Mai
    4438: 2025,   # The Legend of Zelda
    4440: 2025,   # Divanation
    4474: 2025,   # Cold Warrior
    4481: 2025,   # Untitled Hunchback of Notre Dame Project
    4483: 2022,   # Feria: The Darkest Light
    4498: 2025,   # Sandkings
    4516: 2025,    # Here Comes the Flood
    4529: 2025,   # Untitled Marlene Dietrich Netflix Project
    4531: 2025,   # Failsafe
    4534: 2025,   # Button Man: The Killing Game
    4539: 2022,   # Robbing Mussolini
    4543: 2025,   # The Other Mrs
    4554: 2022,   # Another Self
    4562: 2025,   # Untitled Lupita Nyong'o Rihanna Project
    4572: 2025,   # Death Note 2
    4574: 2022,   # Sicak kafa
    4580: 2025,   # Tell Me Everything
    4617: 2022,   # Las niñas de cristal
    4634: 2025,   # Terminator
    4640: 2025,   # Hope
    4650: 2025,   # Untitled Taika Waititi Netflix Roald Dahl Project
    4655: 2025,   # The Seven Deaths of Evelyn Hardcastle
    4669: 2025,   # Assassin's Creed
    4672: 2025,   # One Hundred Years of Solitude
    4682: 2025,    # Untitled Pokémon Project
    4745: 2025,   # Untitled Sister Comedy Netflix Project
    4770: 2025,   # Recursion
    4775: 2025,   # Yu Yu Hakusho
    4788: 2025,   # Untitled GameStop stock market Film
    4790: 2022,   # Santo
    4802: 2025,   # Old Man's War
    4803: 2021,   # W lesie dzis nie zasnie nikt 2
    4815: 2025,   # Captain Laserhawk: A Blood Dragon Remix
    4816: 2022,   # Thar
    4828: 2025,   # Lotería
    4830: 2025,   # Untitled Animated Adam Sandler Project
    4833: 2025,   # Charlie and the Chocolate Factory
    4863: 2025,   # Tomb Raider
    4878: 2022,   # Baruca
    4898: 2025,   # Untitled Charlie Brooker Netflix Project
    4901: 2025,    # Fool Me Once
    3163:2015,
    3161:2024,
    3158:2020,
    3144:2024,
    3134:2022,
    3040:2023,
    3019:1978,
    2981:2024,
    2974:2010,
    2971:2021,
    2779:2022,
    2753:2019,
    2859:2022,
    2905:2024,
    2917:2022,
    2919:2023,
    2947:2022,
    2948:2022,
    2948:2022,
    2959:2022,
    2753:2024,
    2779:2022,
    2804:2022,
    2832:2021,
    2846:2023,
    2522:2022,   
    2584:2022,
    2585:2022,
    2610:1991,
    2632:2022,
    2676:2022,
    2686:2025,
    2695:2022,
    2801:2022,
    2410:2022,
    2450:2023,
    2466:2023,
    2494:2022,
    2506:2021,
    2397:2022,
    2373:2022,
    2362:2022,
    2359:2023,
    2347:2023,
    2295:2023,
    2259:2015,
    2215:2015,
    2177:2025,
    2175:2004,
    2164:2022,
    2126:2022,
    2124:1978,
    2122:2023,
    2091:2021,
    2089:2024,
    2069:2022,
    2055:2023,
    2023:2024,
    1982:2011,
    1916:2024,
    1914:1999,
    1897:2023,
    1896:2022,
    1863:2022,
    1859:2022,
    1841:2024,
    1818:2024,
    1802:2024,
    1800:2025,
    1774:2022,
    1766:1988,
    1765:2025,
    1763:2020,
    1703:2023,
    1684:2022,
    1664:2024,
    1658:2024,
    1645:2024,
    1623:2023,
    1587:2022,
    1530:2024,
    1527:2022,
    1507:2012,
    1494:2022,
    1492:2023,
    1485:2022,
    1464:2019,
    1428:2023,
    1425:2022,
    1419:2005,
    1414:2020,
    1375:2002,
    1165:2022,
    1155:2024,
    878:2022

}

# Loop through dictionary and assign year
for idx, year in movie_years.items():
    Movie_Data.loc[idx, 'YEAR'] = year


In [208]:
Movie_Data['MOVIES']=Movie_Data['MOVIES'].str.replace(r'[+/\\-]',' ',regex=True)

In [209]:
Movie_Data['GENRE'] = Movie_Data['GENRE'].str.replace(r'[\n]','',regex=True)

In [221]:
Movie_Data['ONE-LINE'] = Movie_Data['ONE-LINE'].str.replace("\n","")

In [222]:
Movie_Data['ONE-LINE'] = Movie_Data['ONE-LINE'].str.strip()

In [223]:
Movie_Data.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Gross'],
      dtype='object')

In [224]:
Movie_Data['STARS']=Movie_Data['STARS'].str.replace("\n","")

In [225]:
Movie_Data['STARS']

0           Director:Peter Thorwarth|     Stars:Peri B...
1                       Stars:Chris Wood, Sarah Michel...
2                       Stars:Andrew Lincoln, Norman R...
3                       Stars:Justin Roiland, Chris Pa...
4           Director:Matthias Schweighöfer|     Stars:...
                              ...                        
9563        Director:Nicolai Rohde|     Stars:Felix Kl...
9564                                                     
9565        Director:Orlando von Einsiedel|     Star:P...
9566        Director:Jovanka Vuckovic|     Stars:Morga...
9567        Director:Jovanka Vuckovic|     Stars:Morga...
Name: STARS, Length: 9568, dtype: object

In [226]:
Movie_Data['YEAR'].isnull().sum()

np.int64(0)

In [227]:
Movie_Data.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Gross'],
      dtype='object')

In [228]:
Movie_Data["RATING"].isnull().sum()

np.int64(1400)

In [229]:
Movie_Data["RATING"]

0       6.1
1       5.0
2       8.2
3       9.2
4       NaN
       ... 
9563    NaN
9564    NaN
9565    NaN
9566    NaN
9567    NaN
Name: RATING, Length: 9568, dtype: float64

In [230]:
Movie_Data["MOVIES"].isnull().sum()

np.int64(0)

In [231]:
Movie_Data.dtypes

MOVIES       object
YEAR         object
GENRE        object
RATING      float64
ONE-LINE     object
STARS        object
VOTES        object
RunTime     float64
Gross        object
dtype: object

In [234]:
GENRE_DATE = {
    2124:"adjacent to true crime",
    3144:"Action, Spy, Suspense, Thriller"
}
for idex,genre in GENRE_DATE.items():
    Movie_Data.loc[idex,"GENRE"] = genre

#### Movies Column Total Clean

In [235]:
Movie_Data["MOVIES"].sample(20)

7138                     Comedians in Cars Getting Coffee
472     Kôdo giasu   Hangyaku no rurûshu: Code Geass  ...
5833                                               Fix Us
4618         ReMastered: Tricky Dick and the Man in Black
144                        Jumanji: Welcome to the Jungle
6142                                  La Sangre del Gallo
6759                                             Ultraman
3350                  WISH YOU: Your Melody from My Heart
7486                                     Jupiter's Legacy
7653                                     The Last Kingdom
5233                               Les étoiles vagabondes
575                                           Steins;Gate
6811                                             Lunatics
5281                                           Super Zach
132                                          The Departed
4527                    ReMastered: Who Shot the Sheriff?
1559                                                Stree
4086    The Ha

In [236]:
Movie_Data["MOVIES"] = Movie_Data["MOVIES"].str.title()

#### Working On Rating Column

In [237]:
Movie_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9568 entries, 0 to 9567
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9568 non-null   object 
 1   YEAR      9568 non-null   object 
 2   GENRE     9492 non-null   object 
 3   RATING    8168 non-null   float64
 4   ONE-LINE  9568 non-null   object 
 5   STARS     9568 non-null   object 
 6   VOTES     8168 non-null   object 
 7   RunTime   7008 non-null   float64
 8   Gross     460 non-null    object 
dtypes: float64(2), object(7)
memory usage: 672.9+ KB


In [238]:
Movie_Data["RATING"].isnull().sum()

np.int64(1400)

In [240]:
Movie_Data["RATING"].mean()

np.float64(6.919698824681684)

In [241]:
RATING_MEDIAN = Movie_Data["RATING"].median()

In [242]:
Movie_Data.fillna({"RATING":RATING_MEDIAN},inplace=True)

In [243]:
Movie_Data["RATING"].isna().sum()

np.int64(0)

In [245]:
Movie_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9568 entries, 0 to 9567
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9568 non-null   object 
 1   YEAR      9568 non-null   object 
 2   GENRE     9492 non-null   object 
 3   RATING    9568 non-null   float64
 4   ONE-LINE  9568 non-null   object 
 5   STARS     9568 non-null   object 
 6   VOTES     8168 non-null   object 
 7   RunTime   7008 non-null   float64
 8   Gross     460 non-null    object 
dtypes: float64(2), object(7)
memory usage: 672.9+ KB


In [246]:
Movie_Data["STARS"].sample(20)

8596        Director:Oliver Driver|     Stars:Rorrie D...
5452                    Stars:Adesua Etomi-Wellington,...
3550        Director:Aaron Lieber|     Stars:Alana Bla...
3679        Director:Kivanç Baruönü|     Stars:Gupse O...
9072        Director:Ken Whittingham|     Stars:Jamie ...
5411        Director:Eddie Mullins|     Stars:Leo Fitz...
302                     Stars:Luna Wedler, Jessica Sch...
6409        Director:Steve Petersen|     Stars:Johnny ...
1884                    Stars:Lee Min-ki, Jung So-Min,...
1999                    Stars:Cindy Robinson, Valerie ...
2887        Director:Kheiron|     Stars:Kheiron, Cathe...
6999                    Stars:Emmanuel Esparza, Essine...
1247        Director:Jason Cabell|     Stars:Nicolas C...
49          Director:Zack Snyder|     Stars:Dave Bauti...
5909        Director:Lancelot Oduwa Imasuen|     Stars...
8791        Director:Phil Allora|     Stars:Rachel Egg...
9349                                                     
8038        Di

In [247]:
Movie_Data['DIRECTOR'] = Movie_Data['STARS'].where(Movie_Data['STARS'].str.contains(r'\|')).str.split(r'\|', n=1).str[0]


In [248]:
Movie_Data['DIRECTOR'].sample(10)

5506                         Director:Shawn Arranha
1175                                            NaN
9329                           Director:Jun-hee Han
2032                  Director:Naseef Yusuf Izuddin
5372                          Director:Neville Shah
6603        Directors:Noriyuki Abe, Kazunori Mizuno
8708                           Director:Philip Koch
1361                                            NaN
1384                            Director:Sean Baker
7998                                            NaN
Name: DIRECTOR, dtype: object

In [249]:
Movie_Data["STARS"].sample(10)

3744        Director:Lasja Fauzia|     Stars:Jefri Nic...
5388        Directors:Matt Danner, Dan Fraga|     Star...
2599        Director:Hannes Stöhr|     Stars:Paul Kalk...
4265        Director:Gustavo Hernández|     Stars:Adri...
1899                    Stars:Grace Beedie, Dakota Tay...
7919        Directors:Robert Kenner, Taki Oldham|     ...
3701                    Stars:Flower the Meerkat, Moza...
5978            Director:Jay Chapman|     Star:Todd Glass
8480        Director:Greg Sullivan|     Stars:Ashleigh...
2375        Director:Alexandre Espigares|     Stars:Ra...
Name: STARS, dtype: object

In [250]:
Movie_Data['STARS'] = Movie_Data['STARS'].str.split(r'\|', n=1).str[1].fillna(Movie_Data['STARS'])


In [251]:
Movie_Data['STARS'] = Movie_Data['STARS'].str.replace(r'^\s*Stars:\s*', '', regex=True)


In [252]:
Movie_Data["STARS"].str.title()

0       Peri Baumeister, Carl Anton Koch, Alexander Sc...
1       Chris Wood, Sarah Michelle Gellar, Lena Headey...
2       Andrew Lincoln, Norman Reedus, Melissa Mcbride...
3       Justin Roiland, Chris Parnell, Spencer Grammer...
4       Matthias Schweighöfer, Nathalie Emmanuel, Ruby...
                              ...                        
9563    Felix Klare, Romina Küper, Anna Maria Mühe, Ro...
9564                                                     
9565                                    Star:Prince Harry
9566    Morgan Taylor Campbell, Iñaki Godoy, Rhianna J...
9567    Morgan Taylor Campbell, Jennifer Cheon Garcia,...
Name: STARS, Length: 9568, dtype: object

In [253]:
Movie_Data["STARS"].sample(10)

2501    Daym Drops, Sonya Beckham, Teresa Cottrell, Na...
260     Elijah Wood, Ian McKellen, Viggo Mortensen, Or...
2622    Inori Minase, Kana Hanazawa, Yuka Iguchi, Saor...
4341                           Star:Paco Ignacio Taibo II
5105    Tom Sizemore, Stacy Keach, Kevin P. Farley, Hé...
2913    Bishop Agapit, Catherine Ashton, Serhii Averch...
4532    Maureen O'Hara, Alex Nicol, William Bishop, Ro...
247     Kent Williams, Iemasa Kayumi, Vic Mignogna, Ro...
8692    Colin O'Donoghue, David Bradley, James Faulkne...
7858    Carl Malapa, Némo Schiffman, Manon Bresch, Cor...
Name: STARS, dtype: object

In [254]:
Movie_Data["STARS"].isna().sum()

np.int64(0)

In [255]:
Movie_Data['DIRECTOR'] = Movie_Data['DIRECTOR'].str.replace(r'^\s* Director:\s*', '', regex=True)

In [256]:
Movie_Data["DIRECTOR"].str.title()

0             Peter Thorwarth
1                         NaN
2                         NaN
3                         NaN
4       Matthias Schweighöfer
                ...          
9563            Nicolai Rohde
9564                      NaN
9565    Orlando Von Einsiedel
9566         Jovanka Vuckovic
9567         Jovanka Vuckovic
Name: DIRECTOR, Length: 9568, dtype: object

In [257]:
Movie_Data["DIRECTOR"].sample(10)

7498         Joss Agnew
3292    César Rodrigues
8737       Salva Romero
3021     Stanley Nelson
8626    Bartosz Konopka
1462        Jeff Wadlow
8947       Helen Shaver
5968                NaN
5900       Tekin Girgin
7559     Fabrice Gobert
Name: DIRECTOR, dtype: object

In [258]:
Movie_Data["DIRECTOR"].isna().sum()

np.int64(3692)

In [259]:
Movie_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9568 entries, 0 to 9567
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9568 non-null   object 
 1   YEAR      9568 non-null   object 
 2   GENRE     9492 non-null   object 
 3   RATING    9568 non-null   float64
 4   ONE-LINE  9568 non-null   object 
 5   STARS     9568 non-null   object 
 6   VOTES     8168 non-null   object 
 7   RunTime   7008 non-null   float64
 8   Gross     460 non-null    object 
 9   DIRECTOR  5876 non-null   object 
dtypes: float64(2), object(8)
memory usage: 747.6+ KB


In [260]:
Movie_Data["VOTES"]

0        21,062
1        17,870
2       885,805
3       414,849
4           NaN
         ...   
9563        NaN
9564        NaN
9565        NaN
9566        NaN
9567        NaN
Name: VOTES, Length: 9568, dtype: object

In [261]:
Movie_Data["VOTES"] = Movie_Data["VOTES"].str.replace(",","").astype(float)

In [262]:
Movie_Data["VOTES"] = Movie_Data["VOTES"].astype("Int64")

In [264]:
moive_vote = int(Movie_Data["VOTES"].median())

In [265]:
Movie_Data["VOTES"].fillna(moive_vote,inplace=True)

In [266]:
Movie_Data["VOTES"].isna().sum()

np.int64(0)

In [267]:
Movie_Data["RunTime"].isna().sum()

np.int64(2560)

In [270]:
Movie_Data["RunTime"].min()

np.float64(1.0)

In [271]:
Movie_Data["RunTime"].sample(10)

6423      NaN
6365      NaN
3644     40.0
6149     85.0
7642      NaN
3820    139.0
1721    125.0
2275     97.0
2380     25.0
2567      NaN
Name: RunTime, dtype: float64

In [272]:
Movie_Data.describe()

RATING         VOTES      RunTime
count  9568.000000        9568.0  7008.000000
mean      6.946081  13044.353783    68.874144
std       1.128978  64966.070182    47.278890
min       1.100000           5.0     1.000000
25%       6.400000        225.75    37.000000
50%       7.100000         792.0    60.000000
75%       7.600000        2903.0    95.000000
max       9.900000     1713028.0   853.000000

In [273]:
Movie_Data[Movie_Data["RunTime"] == Movie_Data["RunTime"].min()]

MOVIES  YEAR  \
2854                                     Horsin' Around  2014   
2954               Trollhunters: Happy Birthday To You!  2017   
3378  13 Reasons Why: Season 2 Date Announcement Com...  2018   
3977  Miraculous: Ladybug & Cat Noir: Happy Birthday...  2017   
4129                                        Piggy Tales  2014   
5358                    Pokémon: Happy Birthday To You!  2017   
5446             My Little Pony: Happy Birthday To You!  2017   
5682                     Barbie: Happy Birthday To You!  2017   
5854  Lego Ninjago: Masters Of Spinjitzu: Happy Birt...  2017   
5872                  Beat Bugs: Happy Birthday To You!  2017   
5888       All Hail King Julien: Happy Birthday To You!  2017   
5891                   Dinotrux: Happy Birthday To You!  2017   
6098         Skylanders Academy: Happy Birthday To You!  2017   
6385               Lego Friends: Happy Birthday To You!  2017   
8027                                       Trollhunters  2016   
9167                                       Trollhunters  2016   
9168                           3Below: Tales Of Arcadia  2018   

                                         GENRE  RATING  \
2854          Short, Comedy, Music                 7.4   
2954              Animation, Short                 7.5   
3378                  Short, Drama                 7.2   
3977              Animation, Short                 7.3   
4129  Animation, Adventure, Comedy                 7.2   
5358              Animation, Short                 5.4   
5446              Animation, Short                 7.1   
5682              Animation, Short                 4.1   
5854      Animation, Short, Action                 5.9   
5872              Animation, Short                 7.1   
5888              Animation, Short                 4.4   
5891              Animation, Short                 7.1   
6098              Animation, Short                 5.2   
6385              Animation, Short                 3.6   
8027  Animation, Action, Adventure                 8.3   
9167  Animation, Action, Adventure                 8.1   
9168  Animation, Action, Adventure                 7.6   

                                               ONE-LINE  \
2854  The opening title sequence to the spoof 90s si...   
2954  What's better than a birthday? How about a cho...   
3378                                         Add a Plot   
3977  Break dancing, dragons, balloons and a DJ: Lad...   
4129         A cartoon series of green cartoon piggies.   
5358  Today is your big day, and Ash, Pikachu and al...   
5446  Did someone say "birthday"? Fluttershy and Pin...   
5682  Get ready to make a wish and have the most ama...   
5854  LEGO Ninjago heroes Kai, Zane, Jay and Cole wa...   
5872  They say it's your birthday - and there's no o...   
5888  Nobody parties like King Julien. And today he'...   
5891  Ty and the rest of the Dinotrux just found out...   
6098  The Skylanders are working on their most impor...   
6385  Surprise. The gang's all here to celebrate you...   
8027  Trolls, goblins, humans, Gumm-Gumms -- all man...   
9167  Trolls, humans, gnomes and aliens alike all ma...   
9168  Trolls, humans, gnomes and aliens alike all ma...   

                                                  STARS  VOTES  RunTime Gross  \
2854                                                       627      1.0   NaN   
2954                                                        24      1.0   NaN   
3378  Alisha Boe, Tommy Dorfman, Katherine Langford,...    197      1.0   NaN   
3977                                                        34      1.0   NaN   
4129  Antti Pääkkönen, Antti Pääkkönen, Douglas Blac...    110      1.0   NaN   
5358                                                        21      1.0   NaN   
5446                                                       792      1.0   NaN   
5682                        Ritesh Rajan, America Young      8      1.0   NaN   
5854                                     

In [274]:
Movie_Data[Movie_Data["RunTime"].isna()]

MOVIES  YEAR  \
4                            Army Of Thieves  2021   
13                                  Sex Life  2021   
24    He Man And The Masters Of The Universe  2021   
48                               Sweet Tooth  2021   
76                           Shadow And Bone  2021   
...                                      ...   ...   
9563                               Totenfrau  2022   
9564                                  Arcane  2021   
9565                       Heart Of Invictus  2022   
9566                          The Imperfects  2021   
9567                          The Imperfects  2021   

                                         GENRE  RATING  \
4            Action, Crime, Horror                 7.1   
13          Comedy, Drama, Romance                 5.4   
24    Animation, Action, Adventure                 7.1   
48        Action, Adventure, Drama                 7.9   
76        Action, Adventure, Drama                 7.7   
...                                        ...     ...   
9563               Drama, Thriller                 7.1   
9564  Animation, Action, Adventure                 7.1   
9565            Documentary, Sport                 7.1   
9566     Adventure, Drama, Fantasy                 7.1   
9567     Adventure, Drama, Fantasy                 7.1   

                                               ONE-LINE  \
4     A prequel, set before the events of Army of th...   
13    A suburban mother of two takes a fantasy-charg...   
24    Eternia's Prince Adam discovers the power of G...   
48    A boy who is half human and half deer survives...   
76    Dark forces conspire against orphan mapmaker A...   
...                                                 ...   
9563                                         Add a Plot   
9564                                         Add a Plot   
9565                                         Add a Plot   
9566                                         Add a Plot   
9567                                         Add a Plot   

                                                  STARS  VOTES  RunTime Gross  \
4     Matthias Schweighöfer, Nathalie Emmanuel, Ruby...    792      NaN   NaN   
13    Sarah Shahi, Mike Vogel, Adam Demos, Margaret ...  12172      NaN   NaN   
24                                                         792      NaN   NaN   
48    Nonso Anozie, Christian Convery, Adeel Akhtar,...  37142      NaN   NaN   
76    Jessie Mei Li, Ben Barnes, Archie Renaux, Fred...  60283      NaN   NaN   
...                                                 ...    ...      ...   ...   
9563  Felix Klare, Romina Küper, Anna Maria Mühe, Ro...    792      NaN   NaN   
9564                                                       792      NaN   NaN   
9565                                  Star:Prince Harry    792      NaN   NaN   
9566  Morgan Taylor Campbell, Iñaki Godoy, Rhianna J...    792      NaN   NaN   
9567  Morgan Taylor Campbell, Jennifer Cheon Garcia,...    792      NaN   NaN   

                   DIRECTOR  
4     Matthias Schweighöfer  
13                      NaN  
24                      NaN  
48                      NaN  
76                      NaN  
...                     ...  
9563          Nicolai Rohde  
9564                    NaN  
9565  Orlando von Einsiedel  
9566       Jovanka Vuckovic  
9567       Jovanka Vuckovic  

[2560 rows x 10 columns]

In [275]:
Runtime_median =  Movie_Data["RunTime"].median()

In [276]:
Movie_Data["RunTime"].fillna(Runtime_median,inplace=True)

In [277]:
Movie_Data["RunTime"].isna().sum()

np.int64(0)

In [278]:
Movie_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9568 entries, 0 to 9567
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9568 non-null   object 
 1   YEAR      9568 non-null   object 
 2   GENRE     9492 non-null   object 
 3   RATING    9568 non-null   float64
 4   ONE-LINE  9568 non-null   object 
 5   STARS     9568 non-null   object 
 6   VOTES     9568 non-null   Int64  
 7   RunTime   9568 non-null   float64
 8   Gross     460 non-null    object 
 9   DIRECTOR  5876 non-null   object 
dtypes: Int64(1), float64(2), object(7)
memory usage: 757.0+ KB


In [279]:
Movie_Data["Gross"].isna().sum()

np.int64(9108)

In [280]:
Movie_Data["Gross"].sample(10)

2301    NaN
4877    NaN
7103    NaN
4123    NaN
1468    NaN
7820    NaN
7810    NaN
1973    NaN
2619    NaN
9378    NaN
Name: Gross, dtype: object

In [281]:
def convert_money(x):
    x = str(x).replace("$", "").replace(",", "")
    
    if "M" in x:
        return float(x.replace("M", "")) * 1_000_000
    
    return float(x)

Movie_Data["Gorss_Money"] = Movie_Data["Gross"].apply(convert_money)


In [282]:
movie_price = Movie_Data["Gorss_Money"].median()

In [287]:
Movie_Data["Gorss_Money"].fillna(movie_price,inplace=True)

C:\Users\sitan\AppData\Local\Temp\ipykernel_32728\2534165027.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  Movie_Data["Gorss_Money"].fillna(movie_price,inplace=True)


In [292]:
Movie_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9568 entries, 0 to 9567
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   MOVIES       9568 non-null   object 
 1   YEAR         9568 non-null   object 
 2   GENRE        9492 non-null   object 
 3   RATING       9568 non-null   float64
 4   ONE-LINE     9568 non-null   object 
 5   STARS        9568 non-null   object 
 6   VOTES        9568 non-null   Int64  
 7   RunTime      9568 non-null   float64
 8   Gross        460 non-null    object 
 9   DIRECTOR     5876 non-null   object 
 10  Gorss_Money  9568 non-null   float64
dtypes: Int64(1), float64(3), object(7)
memory usage: 831.7+ KB


In [297]:
Movie_Data.to_csv("Clean_Movie_Data.csv",index=False)